source/modulo_components/test/cpp/test_component_interface.cpp

```diff
@@ -315,16 +313,9 @@ TYPED_TEST(ComponentInterfaceTest, RaiseError) {
 
 TYPED_TEST(ComponentInterfaceTest, AddTrigger) {
   EXPECT_NO_THROW(this->component_->add_trigger("trigger"));
-  ASSERT_FALSE(this->component_->triggers_.find("trigger") == this->component_->triggers_.end());
-  EXPECT_FALSE(this->component_->triggers_.at("trigger"));
-  EXPECT_FALSE(this->component_->get_predicate("trigger"));
+  ASSERT_FALSE(
+      std::find(this->component_->triggers_.cbegin(), this->component_->triggers_.cend(), "trigger")
+      == this->component_->triggers_.cend());
   EXPECT_NO_THROW(this->component_->trigger("trigger"));
-  // When reading, the trigger will be true only once
-  this->component_->triggers_.at("trigger") = true;
-  EXPECT_TRUE(this->component_->triggers_.at("trigger"));
-  EXPECT_TRUE(this->component_->get_predicate("trigger"));
-  // After the predicate function was evaluated once, the trigger is back to false
-  EXPECT_FALSE(this->component_->triggers_.at("trigger"));
-  EXPECT_FALSE(this->component_->get_predicate("trigger"));
 }
```

Comment:
I can't see how the triggering still works. Triggers are never added to the actual predicates_ map, are just published as true on `trigger()`, but I don't see where or how it would be published as false after being triggered. We normally waiting a step to put the trigger back down to false, but maybe you can just try 

```c++
  publish_predicate(trigger_name, true);
  publish_predicate(trigger_name, false);
```

Reply:
You are right, I did make this too easy. See my most recent commit. I believe that with
```
  this->set_predicate(trigger_name, true);
  this->predicates_.at(trigger_name).set_predicate([]() { return false; });
```
It will work because the first line changes the predicate value and fires it immediately and then the second step changes the internal predicate value back to false again such that the next time we check all predicate values, it is a falling edge.